### DISCUSSION TITLE
Get running pyscript version number

### DISCUSSION THREAD

**Discussion by deepcoder** [custom-components/pyscript#503] [Q&A]
I'm a noob here, I'm sure I am missing the simple answer to this:

How do it 'introspect' on pyscript and get the version number currently running?

Thanks!

**Comment by craigbarratt** [comment]
I don't think there is a way to determine the `pyscript` version from a run-time script, unless it can be extracted from HASS or HACS internals.

Is there a need or application to get the version?  I'm open to adding that if there is a good reason.

**Comment by deepcoder** [comment]
As I said I am noob. I am finding pyscript an interesting tool, thanks. 

So, I'm not sure this is a major need. However, I did just upgrade to your 1.5 version and noted it has some changed behaviors from prior versions. I am not to the level of offering any public integrations that are based on pyscript, however, it does seem to be a good platform for some public offerings. And as I read / listen to the ever increasing complex/longtail upgrade timeline/timeframe that users are doing, knowing what works and does not work with the version of Home Assistant and integrations/tools that I am running is a useful feature.

Again, thanks for your work on this project!

**Comment by ALERTua** [comment]
Basically, HACS pulls addons revisions from GitHub. So if you pulled a release version of pyscript (say, 1.5.0) — you are at this release version. But nothing stops you from modifying files in your locally downloaded pulled copy. Would this be the other version? Nope. Anyhow this is a bunch of files. So there's only a "version" of the HACS addon if you haven't modified any of its files. a version that corresponds to the revision pulled. If this revision is tagged with a release version - you can call it with the tag name (e.g. 1.5.0), but if you chose to just download a branch, this would be a commit hash as a "version".
So if you want o report a bug, you can report it as a version bug, if you downloaded a release version, or the bug should be related to a commit you are on, if not.
